R/HDF5-write.R

```diff
@@ -439,3 +441,50 @@ hdf5_path_exists <- function(file, target_path) {
 
   target_path %in% paths
 }
+
+#' Write H5AD
+#'
+#' Write an H5AD file
+#'
+#' @param object The object to write, either a "SingleCellExperiment" or a
+#' "Seurat" object
+#' @param path Path of the file to write to
+#'
+#' @return `path` invisibly
+#' @export
+#'
+#' @examples
+#' # Write a SingleCellExperiment as a H5AD
+#' h5ad_file <- tempfile(fileext = ".h5ad")
+#' if (requireNamespace("SingleCellExperiment", quietly = TRUE)) {
+#'   sce <- dummy_data(output = "SingleCellExperiment")
+#'   write_h5ad(sce, h5ad_file)
+#' }
+#'
+#' # Write a Seurat as a H5AD
+#' h5ad_file <- tempfile(fileext = ".h5ad")
+#' if (requireNamespace("SeuratObject", quietly = TRUE)) {
+#'   seurat <- dummy_data(output = "Seurat")
+#'   write_h5ad(seurat, h5ad_file)
+#' }
+write_h5ad <- function(object, path) {
+  if (is(object, "SingleCellExperiment")) {
+    from_SingleCellExperiment(
+      object,
+      output_class = "HDF5AnnData",
+      file = path
+    )
+  } else if (is(object, "Seurat")) {
+    from_Seurat(
+      object,
+      output_class = "HDF5AnnData",
+      file = path
+    )
+  } else {
```

Comment:
also allow writing object which inherit `AbstractAnnData`

```suggestion
  } else if (inherits(object, "AbstractAnnData")) {
    to_HDF5AnnData(object, path) # todo: implement
  }